core/src/main/java/org/apache/iceberg/ReachableFileCleanup.java

```diff
@@ -84,19 +77,74 @@ public void cleanFiles(TableMetadata beforeExpiration, TableMetadata afterExpira
     deleteFiles(manifestListsToDelete, "manifest list");
   }
 
-  private Set<ManifestFile> readManifests(Set<Snapshot> snapshots) {
-    Set<ManifestFile> manifestFiles = Sets.newHashSet();
-    for (Snapshot snapshot : snapshots) {
-      try (CloseableIterable<ManifestFile> manifestFilesForSnapshot = readManifestFiles(snapshot)) {
-        for (ManifestFile manifestFile : manifestFilesForSnapshot) {
-          manifestFiles.add(manifestFile.copy());
-        }
-      } catch (IOException e) {
-        throw new RuntimeIOException(
-            e, "Failed to close manifest list: %s", snapshot.manifestListLocation());
-      }
+  // Returns the current set of manifests while pruning the manifests to delete
+  // If all candidate manifests are pruned the returned set of current manifests will be empty
+  private Set<ManifestFile> findCurrentManifestsAndPruneCandidates(
+      Set<Snapshot> snapshots, Set<ManifestFile> deletionCandidates) {
+    Set<ManifestFile> currentManifests = ConcurrentHashMap.newKeySet();
+    if (deletionCandidates.isEmpty()) {
```

Comment:
I think the caller should check for this. If the candidate set is empty before calling this, then the whole operation should short circuit and return, not just this operation. That avoids the awkwardness of getting an empty set of `currentManifests` that is probably not correct.

Reply:
makes sense, I updated so that finding manifests and is only done if there are deletion candidates, I definitely think it's cleaner that way.